packages/orbit-components/src/tmp-tailwind/ButtonLink/index.tsx

```diff
@@ -0,0 +1,108 @@
+};
+
+const typeCompactStyles: Record<Type, string> = {
+  primary:
+    "text-button-link-primary-foreground focus:text-button-link-primary-foreground-focus active:text-button-link-primary-foreground-active hover:text-button-link-primary-foreground-hover",
+  secondary:
+    "text-button-link-secondary-foreground focus:text-button-link-secondary-compact-foreground-hover active:text-button-link-secondary-compact-foreground-active hover:text-button-link-secondary-compact-foreground-hover",
+  critical:
+    "text-button-link-critical-foreground focus:text-button-link-critical-foreground-focus active:text-button-link-critical-foreground-active hover:text-button-link-critical-foreground-hover",
+};
+
+const typeDisabledStyles: Record<Type, string> = {
+  primary: "bg-button-link-primary-background text-button-link-primary-foreground",
+  secondary: "bg-button-link-secondary-background text-button-link-secondary-foreground",
+  critical: "bg-button-link-critical-background text-button-link-critical-foreground",
+};
+
+const typeCompactDisabledStyles: Record<Type, string> = {
+  primary: "text-button-link-primary-foreground",
+  secondary: "text-button-link-secondary-foreground",
+  critical: "text-button-link-critical-foreground",
+};
+
+const ButtonLink = React.forwardRef<HTMLButtonElement, Props>(
+  (
+    {
+      type = "primary",
+      size = "normal",
+      compact,
+      disabled,
+      children,
+      iconLeft,
+      iconRight,
+      ...props
+    },
+    ref,
+  ) => {
+    return (
+      <ButtonPrimitive
+        ref={ref}
+        iconLeft={iconLeft}
+        iconRight={iconRight}
+        disabled={disabled}
+        {...props}
+        className={cx(
+          "space-x-xs rtl:space-x-reverse",
+          sizeStyles[size],
+          compact !== true && disabled !== true && typeStyles[type],
+          compact === true && disabled !== true && typeCompactStyles[type],
+          compact !== true && disabled === true && typeDisabledStyles[type],
+          compact === true && disabled === true && typeCompactDisabledStyles[type],
+          compact !== true &&
+            children != null &&
+            iconLeft == null &&
+            iconRight == null &&
+            paddingNoIconsStyles[size],
+          compact !== true &&
+            children != null &&
+            iconLeft != null &&
+            iconRight == null &&
+            paddingLeftIconStyles[size],
+          compact !== true &&
+            children != null &&
+            iconLeft == null &&
+            iconRight != null &&
+            paddingRightIconStyles[size],
+          compact !== true &&
```

Comment:
this condition seems to be repetitive, could you assign it to a variable? 

```
 compact !== true &&
            children != null &&
            ```

Reply:
while this `compact !== true && children != null` bit is used four times, imo it is better to be explicit with these low level conditions than abstracting them away. with these you _immediately_ know they're props and what they represent

if we added _another_ variable that is composed of this, e.g. `isNotCompactWithoutChildren`, that is almost as many characters as the explicit condition 😅 and you have more cognitive load to bear

Reply:
@oreqizer You may try smth like that, looks shortened and more readable imo 
```
    const isIconPaddingStyle = compact !== true && children != null;

    return (
      <ButtonPrimitive
        ref={ref}
        iconLeft={iconLeft}
        iconRight={iconRight}
        disabled={disabled}
        {...props}
        className={cx(
          "space-x-xs rtl:space-x-reverse",
          sizeStyles[size],
          compact === true && [
            disabled === true typeDisabledStyles[type],
            disabled !== true && typeCompactStyles[type],
          ],
          compact !== true && [
            disabled === true && typeDisabledStyles[type],
            disabled !== true && typeStyles[type],
          ],
          isIconPaddingStyle && [
            iconLeft == null &&
            iconRight == null &&
            paddingNoIconsStyles[size],
            iconLeft != null &&
            iconRight == null &&
            paddingLeftIconStyles[size],
            iconLeft == null &&
            iconRight != null &&
            paddingRightIconStyles[size],
            iconLeft != null &&
            iconRight != null &&
            paddingBothIconsStyles[size]
          ],
          props.circled === true && circledStyles[size],
          props.circled === true && children == null && circledIconOnlyStyles[size],
        )}
      >
        {children}
      </ButtonPrimitive>

```

Reply:
I don't know man 😕 I just very much prefer less words and less nesting in code

I'd understand if it was like 20+ cases, but this is really compact and explicit, even when unrolled

cognitive load here seems much bigger, too. when I see `isIconPaddingStyle` I have _no idea_ where is this coming from. I have to jump up and see what does it represent

Reply:
> while this compact !== true && children != null bit is used four times, imo it is better to be explicit with these low level conditions than abstracting them away. with these you immediately know they're props and what they represent

Not sure if I can agree with that, when there are a lot of lines I feel like am struggling when I am reading that, especially when they look almost the same and there are multiple conditions in one single line and you have them like > 4 rows. Assigning it to a variable you can call "isSomething" and it also says immediately what is it if it's properly named. Plus (as I showed above) maybe we should consider using arrays to group them.

Reply:
aight I can go with the array syntax, without an extra variable though. is that good enoguh of a compromise for you?

Reply:
yep :)